Ubuntu install from source fails at vet

## Expected Behavior

I want to install minio on a fresh Ubuntu server on digitalocean. I followed instructions in README + https://docs.minio.io/docs/how-to-install-golang 

Ubuntu 16.04.2 LTS (GNU/Linux 4.4.0-64-generic x86_64) 2GB ram 2CPU

go env shows:
```
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/root/work"
GORACE=""
GOROOT="/root/go"
GOTOOLDIR="/root/go/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build415224862=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```

## Current Behavior
I get the following with make
```
Checking deps:
Checking project is in GOPATH:
Installing golint:
Installing gocyclo:
Installing deadcode:
Installing misspell:
Installing ineffassign:
Running vet:
cmd/xl-v1.go:143: declaration of "err" shadows declaration at cmd/xl-v1.go:107
Makefile:73: recipe for target 'vet' failed
make: *** [vet] Error 1
```

I did not work with go v1.7.5. I tried with other version v1.7.1, 1.7.2, 1.7.3 or 1.7.4 but they won't work either. 

Has anyone managed to compile minio on Ubuntu? 

I also had the same exact error on macOS but downgraded go from 1.8 to 1.7.5 and it worked. The same trick didn't work on Ubuntu though.

Thanks

> Running vet:
> cmd/xl-v1.go:143: declaration of "err" shadows declaration at cmd/xl-v1.go:107
> Makefile:73: recipe for target 'vet' failed
> make: *** [vet] Error 1

This should be something recent not reproducible on my `Ubuntu 16.04`  with go 1.7.5

```
$ go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOOS="linux"
GOPATH="/home/harsha/mygo"
GORACE=""
GOROOT="/home/harsha/.gimme/versions/go1.7.5.linux.amd64"
GOTOOLDIR="/home/harsha/.gimme/versions/go1.7.5.linux.amd64/pkg/tool/linux_amd64"
CC="gcc"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build526592543=/tmp/go-build -gno-record-gcc-switches"
CXX="g++"
CGO_ENABLED="1"
```

```
$ make
Checking deps:
Checking project is in GOPATH:
Installing golint:
Installing gocyclo:
Installing deadcode:
Installing misspell:
Installing ineffassign:
Running vet:
Running fmt:
Running lint:
Running cyclo:
Installing minio:
```

--- a/Makefile
+++ b/Makefile
@@ -71,10 +71,8 @@ verifiers: vet fmt lint cyclo spelling
 
 vet:
 	@echo "Running $@:"
-	@go tool vet -all ./cmd
-	@go tool vet -all ./pkg
-	@go tool vet -shadow=true ./cmd
-	@go tool vet -shadow=true ./pkg
+	@go vet github.com/minio/minio/cmd/...
+	@go vet github.com/minio/minio/pkg/...
 
 fmt:
 	@echo "Running $@:"

--- a/cmd/xl-v1.go
+++ b/cmd/xl-v1.go
@@ -140,9 +140,9 @@ func newXLObjects(storageDisks []StorageAPI) (ObjectLayer, error) {
 	// Check if object cache is enabled.
 	if xl.objCacheEnabled {
 		// Initialize object cache.
-		objCache, err := objcache.New(maxCacheSize, objcache.DefaultExpiry)
-		if err != nil {
-			return nil, err
+		objCache, oerr := objcache.New(maxCacheSize, objcache.DefaultExpiry)
+		if oerr != nil {
+			return nil, oerr
 		}
 		objCache.OnEviction = func(key string) {
 			debug.FreeOSMemory()

